src/rt/test/func/memory/memory_rc.h

```diff
@@ -143,37 +143,125 @@ namespace memory_rc
    **/
   void test_cycles()
   {
-    auto& alloc = ThreadAlloc::get();
-    auto* o = new (alloc) C;
-
-    // Allocate some reachable objects.
-    auto* o1 = new (alloc, o) C;
-    auto* o2 = new (alloc, o) C;
-    auto* o3 = new (alloc, o) C;
-    auto* o4 = new (alloc, o) C;
-    auto* o5 = new (alloc, o) C;
-
-    // cycle: o1 -> o2 -> o3 -> o4 -> o5 -> o1
-    o1->f1 = o2;
-    o2->f1 = o3;
-    o3->f1 = o4;
-    o4->f1 = o5;
-    RegionRc::incref(o1, o);
-    o5->f1 = o1;
-
-    check(Region::debug_size(o) == 6);
-    o5->f1 = nullptr;
-    RegionRc::decref(alloc, o1, o);
-    check(Region::debug_size(o) == 6);
-    // When cycle detection exists. This should dealloc the objects.
-    // check(Region::debug_size(o) == 0);
-
-    // Break the cycle
-    RegionRc::decref(alloc, o1, o);
-    check(Region::debug_size(o) == 1);
-
-    Region::release(alloc, o);
-    snmalloc::debug_check_empty<snmalloc::Alloc::StateHandle>();
+    // Start with a simple cycle
+    {
+      auto& alloc = ThreadAlloc::get();
+      auto* o = new (alloc) C;
+
+      // Allocate some reachable objects.
+      auto* o1 = new (alloc, o) C;
+      auto* o2 = new (alloc, o) C;
+      auto* o3 = new (alloc, o) C;
+      auto* o4 = new (alloc, o) C;
+      auto* o5 = new (alloc, o) C;
+      auto* o6 = new (alloc, o) C;
+
+      // cycle: o6 -> (o1 -> o2 -> o3 -> o4 -> o5 -> o1)
+      o1->f1 = o2;
+      o2->f1 = o3;
+      o3->f1 = o4;
+      o4->f1 = o5;
+      o5->f1 = o1;
+
+      RegionRc::incref(o1, o);
+      o6->f1 = o1;
+
+      check(Region::debug_size(o) == 7);
```

Comment:
Check we don't collect cycles?
```suggestion
      check(Region::debug_size(o) == 7);
      RegionRc::gc_cycles(alloc, o);
      check(Region::debug_size(o) == 7);
```

Reply:
Fixed here: 3433606a494e70ec9cdf03d369f651535196ad0a